org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/DOMAttributeFormatter.java

```diff
@@ -88,8 +89,7 @@ public void formatAttribute(DOMAttr attr, int prevOffset, boolean singleAttribut
 		// --> <a name='value'> </a>
 		String originalValue = attr.getOriginalValue();
 		if (getEnforceQuoteStyle() == EnforceQuoteStyle.preferred && originalValue != null) {
-			if (originalValue.charAt(0) != getQuotationAsChar()
-					&& StringUtils.isQuote(originalValue.charAt(0))) {
+			if (originalValue.charAt(0) != getQuotationAsChar() && StringUtils.isQuote(originalValue.charAt(0))) {
 				formatterDocument.replaceQuoteWithPreferred(attr.getNodeAttrValue().getStart(),

```

Comment:
I believe we can use `createTextEditIfNeeded()` directly here since https://github.com/eclipse/lemminx/blob/07134c571566ac04f3cbc500b870e8bc7ad17dc1/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/services/format/XMLFormatterDocumentNew.java#L372

Reply:
Good catch, I think it is good to keep replaceQuoteWithPreferred to know where quote is managed, but I change the signature and now this method is relevant.